### PR TITLE
Replace requester name referral table

### PR DESCRIPTION
**Pull request by nicolasjourdain**

### CODE REVIEW

File: src/frontend/js/types/index.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export interface Referral {
 
 export interface ReferralLite
   extends Pick<Referral, 'assignees' | 'due_date' | 'id' | 'object' | 'state'> {
-  users: UserLite[];
+  users: User[];
```
Comment (mbenadda) [inline src/frontend/js/types/index.ts:28]: This change seems wrong to me: referral lites are still including only user lites

File: src/frontend/js/translations/en-US.json [hunk 2]
```diff
@@ -202,8 +202,7 @@
   "components.ReferralDetail.progressStep4": "Currently processing",
   "components.ReferralDetail.progressStep5": "Undergoing validation",
   "components.ReferralDetail.progressStep6": "Answer sent",
-  "components.ReferralDetail.requesters": "Requesters",
-  "components.ReferralTable.requesters": "Requesters",
+  "components.ReferralDetail.requesters": "",
```
Comment (mbenadda) [inline src/frontend/js/translations/en-US.json:205]: Why is this now empty?

File: src/frontend/js/translations/fr-FR.json [hunk 3]
```diff
@@ -202,8 +202,7 @@
   "components.ReferralDetail.progressStep4": "En cours de traitement",
   "components.ReferralDetail.progressStep5": "En attente de validation",
   "components.ReferralDetail.progressStep6": "Réponse envoyée",
-  "components.ReferralDetail.requesters": "Demandeurs",
-  "components.ReferralTable.requesters": "Demandeurs",
+  "components.ReferralDetail.requesters": "",
```
Comment (mbenadda) [inline src/frontend/js/translations/fr-FR.json:205]: Why is this now empty?

File: src/frontend/js/data/index.tsx [hunk 4]
```diff
@@ -122,6 +122,7 @@ export type UseReferralLitesParams = {
   unit?: string[];
   user?: string[];
```
Comment (mbenadda) [inline src/frontend/js/data/index.tsx:123]: You have actually removed this parameter from the API so you should also remove it from the type definition.

File: src/frontend/js/components/ReferralTable/index.tsx [hunk 5]
```diff
@@ -297,7 +297,7 @@ export const ReferralTable: React.FC<ReferralTableProps> = ({
                   </th>
                   <td>
                     {referral.users
-                      .map((userLite) => getUserFullname(userLite))
+                      .map((User) => User.unit_name)
```
Comment (mbenadda) [inline src/frontend/js/components/ReferralTable/index.tsx:300]:
```suggestion
                      .map((user) => user.unit_name)
```

File: src/frontend/js/components/ReferralTable/types.tsx [hunk 6]
```diff
@@ -7,7 +7,7 @@ export enum FilterColumns {
   DUE_DATE = 'due_date',
   STATE = 'state',
   UNIT = 'unit',
-  USER = 'user',
+  USER = 'users_unit_name',
```
Comment (mbenadda) [inline src/frontend/js/components/ReferralTable/types.tsx:10]: I think the `USER` key could change to reflect the fact that it's now units and not users we're dealing with

File: src/frontend/js/components/ReferralTable/index.tsx [hunk 7]
```diff
@@ -50,7 +50,7 @@ const messages = defineMessages({
     id: 'components.ReferralTable.object',
   },
   requesters: {
-    defaultMessage: 'Requesters',
+    defaultMessage: "Unit's Requesters",
```
Comment (mbenadda) [inline src/frontend/js/components/ReferralTable/index.tsx:53]:
```suggestion
    defaultMessage: "Requesters' unit(s)",
```

File: src/frontend/js/components/ReferralTable/Filters.tsx [hunk 8]
```diff
@@ -378,7 +378,10 @@ export const Filters = ({
             {filters[FilterColumns.USER]!.map((user) => (
               <div className="tag tag-blue" key={user.id}>
                 <FormattedMessage {...messages[FilterColumns.USER]} />:{' '}
-                {getUserFullname(user)}
+                {
+                  user.unit_name
+                  //getUserFullname(user)
```
Comment (mbenadda) [inline src/frontend/js/components/ReferralTable/Filters.tsx:383]:
```suggestion
```

File: src/backend/partaj/core/api/user.py [hunk 9]
```diff
@@ -49,23 +49,30 @@ def list(self, request, *args, **kwargs):
...
+            queryset = self.queryset.annotate(
+                first_then_last=Concat(F("first_name"), Value(" "), F("last_name")),
+                last_then_first=Concat(F("last_name"), Value(" "), F("first_name")),
+            ).filter(
+                Q(first_then_last__istartswith=query)
+                | Q(last_then_first__istartswith=query)
+                | Q(email__istartswith=query)
+            )
+
+        if search_type == "unit_name":
```
Comment (mbenadda) [inline src/backend/partaj/core/api/user.py:72]: I would put this condition first and then revert to the default search type whether there is not search type or it does not match "unit_name"

File: src/frontend/js/components/ReferralTable/index.tsx (hunk 7 above)
Comment (nicolasjourdain) [reply src/frontend/js/components/ReferralTable/index.tsx:53]: it could be "Requester's unit(s)" ?

File: src/backend/partaj/core/api/user.py [hunk 10]
```diff
@@ -49,23 +49,29 @@ def list(self, request, *args, **kwargs):
...
-        # and last name, and filtering on their first characters (as well as email's).
-        queryset = self.queryset.annotate(
-            first_then_last=Concat(F("first_name"), Value(" "), F("last_name")),
-            last_then_first=Concat(F("last_name"), Value(" "), F("first_name")),
-        ).filter(
-            Q(first_then_last__istartswith=query)
-            | Q(last_then_first__istartswith=query)
-            | Q(email__istartswith=query)
-        )
+        if not search_type:
```
Comment (mbenadda) [inline src/backend/partaj/core/api/user.py:64]:
```suggestion
        else:
```

File: src/frontend/js/components/ReferralTable/index.tsx (hunk 7 above)
Comment (mbenadda) [reply src/frontend/js/components/ReferralTable/index.tsx]: `"Requesters' unit(s)"` means several requesters, `"Requester's unit(s)"` just one requester

File: src/frontend/js/data/index.tsx (hunk 4 above)
Comment (mbenadda) [reply src/frontend/js/data/index.tsx]: This makes sense but then we cannot remove it from the API.

File: src/backend/partaj/core/api/referral_lite.py [hunk 11]
```diff
@@ -101,9 +101,9 @@ def list(self, request, *args, **kwargs):
         if len(unit):
             es_query_filters += [{"terms": {"units": unit}}]
 
-        user = form.cleaned_data.get("user")
```
Comment (mbenadda) [inline src/backend/partaj/core/api/referral_lite.py:104]: Should we not let this one in place and just add `users_unit_name` in addition to it?

**Review by mbenadda — APPROVED** [review]

**Pull request merged**